CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ option(JVM_BINDINGS "Build JVM bindings" OFF)
 option(R_LIB "Build shared library for R package" OFF)
 ## Dev
 option(GOOGLE_TEST "Build google tests" OFF)
+option(USE_DMLC_GTEST "Use google tests bundled with dmlc-core submodule (EXPERIMENTAL)" OFF)
```

Comment:
I thought we are using system level GTest by `find_package`, just a side note there's an env `GTEST_ROOT` can be used: https://cmake.org/cmake/help/v3.12/module/FindGTest.html .

Reply:
I introduced USE_DMLC_GTEST as an optional flag, because I couldn’t get gtest installed as a system lib and have CMake find it. I suspect this may be why a lot of examples on the Internet choose to embed gtest as a CMake submodule

Reply:
Actually, it was me 7 month ago, who added embedded gtest into dmlc-core: dmlc/dmlc-core#468. I recall having lots of trouble getting CMake find gtest on Windows, so hence gtest was added via add_subdirectory().